homeassistant/components/intent/__init__.py

```diff
@@ -61,24 +61,36 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 class OnOffIntentHandler(intent.ServiceIntentHandler):
     """Intent handler for on/off that handles covers too."""
 
-    async def async_call_service(self, intent_obj: intent.Intent, state: State) -> None:
+    async def async_call_service(
+        self, intent_obj: intent.Intent, state: State
+    ) -> tuple[bool | None, State]:
         """Call service on entity with special case for covers."""
         hass = intent_obj.hass
 
         if state.domain == COVER_DOMAIN:
             # on = open
             # off = close
-            await hass.services.async_call(
+            result = await hass.services.async_call(
                 COVER_DOMAIN,
                 SERVICE_OPEN_COVER
                 if self.service == SERVICE_TURN_ON
                 else SERVICE_CLOSE_COVER,
                 {ATTR_ENTITY_ID: state.entity_id},
                 context=intent_obj.context,
+                blocking=True,
+                limit=self.service_timeout,
             )
-        else:
-            # Fall back to homeassistant.turn_on/off
-            await super().async_call_service(intent_obj, state)
+
+            return result, state
```

Comment:
result will be False if it hit the timeout of 2s. I would consider that still assuming it will succeed